pkg/pushsync/pushsync.go

```diff
@@ -168,6 +168,13 @@ func (ps *PushSync) handler(ctx context.Context, p p2p.Peer, stream p2p.Stream)
 				ctxd, canceld := context.WithTimeout(context.Background(), timeToWaitForPushsyncToNeighbor)
 				defer canceld()
 
+				realClosestPeer, err := ps.topologyDriver.ClosestPeer(chunk.Address(), false, swarm.ZeroAddress)
+				if err == nil {
+					if !realClosestPeer.Equal(p.Address) {
+						ps.metrics.TotalReplicationFromDistantPeer.Inc()
```

Comment:
what about having another metric which is the reverse? i.e. have an `else` clause with a replication metric for the number of correct replications we got. also having an extra metric at the top between L165 and L166 to count how many total replication calls we get from other full nodes? this way we can have percentage rates of how many calls were correct, how many calls were not, etc